src/MarkerClusterGroup.jsx

```diff
@@ -0,0 +1,129 @@
+import React, { useContext, useMemo } from 'react'
+import styled, { ThemeContext } from 'styled-components';
+import {
+  createElementObject,
+  createPathComponent,
+  extendContext,
+} from '@react-leaflet/core';
+import L from 'leaflet';
+import 'leaflet.markercluster';
+import ReactDOMServer from 'react-dom/server';
+import { Box, Text } from 'grommet';
+import { statuses } from './GrommetMarker';
+
+const StyledBox = styled(Box)`
+  &:hover {
+    // TO DO revisit "hover" color with designers. Use of "background-layer-overlay" feels strange semantically
+    border: ${props => `${props.theme?.global?.borderSize?.[props.border.size] || props.border.size} solid ${props.theme?.global?.colors?.['background-layer-overlay']}`};
+  }
+`
+const StyledCount = styled(Text)`
+  // TO DO for some reason "Helvetica" was coming but not Metric?
+  font-family: Metric;
+  // multiplier of font-size, for tighter alignment
+  line-height: 1;
+`
+
+// TO DO in production, this threshold should be set to 1000
+// using 10 since our data set is small and I want to demo the
+// different cluster flavors
+const threshold = 10;
+const formatCount = (count) => count > threshold ? `${Math.round(count / threshold)}k` : count;
+
+
+// give critical status priority to be flagged (even if there's only 1 in a cluster of 20+) 
+// this logic can be changed depending on preference decided later.
+// TO DO can this loop be made more efficient?
+const getClusterStatus = (childMarkers) => {
+  // APPROACH A, single while loop where we eject from the loop is a critical status is found
+  let critical = false;
+  let warning = false;
+  let good = false;
+  let index = 0;
+  while (!critical && index <= childMarkers.length - 1) {
+    const status = childMarkers[index].options.icon.options.status;
+    if (status === 'critical') critical = true;
+    else if (status === 'warning') warning = true;
+    else if (status === 'good') good = true;
+    index += 1;
+  }
+
+  if (critical) return 'critical'
+  else if (warning) return 'warning'
+  else if (good) return ('good')
+  else return 'unknown'
```

Comment:
Same efficiency, but a little less verbose. Debating whether unknown should trump good....

```suggestion
let mostSevere = 'good';
let index = 0;
while (mostSevere !== 'critical' && index <= childMarkers.length - 1) {
  const status = childMarkers[index].status
  if (status === 'critical') mostSevere = 'critical';
  else if (status === 'warning') mostSevere = 'warning';
  else if (status === 'unknown') mostSevere = 'unknown';
  index += 1;
}
return mostSevere;
```

Reply:
Done.